helix-core/src/test/java/org/apache/helix/integration/TestCustomizedViewAggregation.java

```diff
@@ -0,0 +1,404 @@
+   * instance    state type  resource    partition            key                 value
+   * ---------------------------------------------------------------------------------
+   *    0            A          A           1            CURRENT_STATE         StatusA1 - D
+   *    0            B          A           1            CURRENT_STATE         StatusB1
+   *    0            B          A           2            CURRENT_STATE         StatusB2  -M -> StatusB3
+   *    0            A          B           2            CURRENT_STATE         StatusA2  -M -> StatusA1
+   *    1            C          A           1            CURRENT_STATE         StatusC1
+   *    1            C          A           2            CURRENT_STATE         StatusC2
+   *    1            A          B           1            CURRENT_STATE         StatusA3 -D
+   *    1            B          B           1            CURRENT_STATE         StatusB3 -D -M-> StatusB2
+   *    1            C          B           1            CURRENT_STATE         StatusC3 -M -> StatusC1
+   *
+   *    -D: to be deleted in the test
+   *    -M: to be modified in the test
+   */
+  @Test
+  public void testCustomizedStateViewAggregation() throws Exception {
+
+    update(INSTANCE_0, CustomizedStateType.TYPE_A, RESOURCE_A, PARTITION_A1,
+        CurrentStateValues.StatusA1);
+    update(INSTANCE_0, CustomizedStateType.TYPE_B, RESOURCE_A, PARTITION_A1,
+        CurrentStateValues.StatusB1);
+    update(INSTANCE_0, CustomizedStateType.TYPE_B, RESOURCE_A, PARTITION_A2,
+        CurrentStateValues.StatusB2);
+    update(INSTANCE_0, CustomizedStateType.TYPE_A, RESOURCE_B, PARTITION_B2,
+        CurrentStateValues.StatusA2);
+    update(INSTANCE_1, CustomizedStateType.TYPE_C, RESOURCE_A, PARTITION_A1,
+        CurrentStateValues.StatusC1);
+    update(INSTANCE_1, CustomizedStateType.TYPE_C, RESOURCE_A, PARTITION_A2,
+        CurrentStateValues.StatusC2);
+    update(INSTANCE_1, CustomizedStateType.TYPE_A, RESOURCE_B, PARTITION_B1,
+        CurrentStateValues.StatusA3);
+    update(INSTANCE_1, CustomizedStateType.TYPE_B, RESOURCE_B, PARTITION_B1,
+        CurrentStateValues.StatusB3);
+
+    // Test batch update API to update several customized states in the same customized state type for one resource, but for now only CURRENT_STATE will be aggregated in customized view
+    Map<String, String> customizedStates = Maps.newHashMap();
+    customizedStates.put("CURRENT_STATE", CurrentStateValues.StatusC3.name());
+    customizedStates.put("PREVIOUS_STATE", CurrentStateValues.StatusC1.name());
+    _customizedStateProvider_participant1
+        .updateCustomizedState(CustomizedStateType.TYPE_C.name(), RESOURCE_B, PARTITION_B1,
+            customizedStates);
+    updateLocalCustomizedViewMap(INSTANCE_1, CustomizedStateType.TYPE_C, RESOURCE_B, PARTITION_B1,
+        CurrentStateValues.StatusC3);
+
+    // Only listen to Type A
+    setTypesToListenInRoutingTableProvider(Arrays.asList(CustomizedStateType.TYPE_A));
```

Comment:
This is not the right way to restrict the customized state types for aggregation. We should change the aggregation config to denote what are the states to be aggregated.

Reply:
Added the changes in aggregation config. Now I am testing both changing aggregation config and routing table provider data sources.